### PR TITLE
add clean triggers for accountResourceRefVO

### PR DESCRIPTION
**Pull request by xuexuemiao**
./runMavenProfile deploydb

```
Successfully applied 15 migrations to schema `zstack` (execution time 00:02.472s).
Flyway 3.2.1 by Boxfuse

Database: jdbc:mysql://localhost:3306/zstack_rest (MySQL 5.5)
Cleaned schema `zstack_rest` (execution time 00:00.014s)
Flyway 3.2.1 by Boxfuse

Database: jdbc:mysql://localhost:3306/zstack_rest (MySQL 5.5)
Validated 1 migration (execution time 00:00.012s)
Creating Metadata table: `zstack_rest`.`schema_version`
Current version of schema `zstack_rest`: << Empty Schema >>
WARNING: outOfOrder mode is active. Migration of schema `zstack_rest` may not be reproducible.
Migrating schema `zstack_rest` to version 0.6 - schema buildin httpserver
Successfully applied 1 migration to schema `zstack_rest` (execution time 00:00.169s).
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 5.647s
[INFO] Finished at: Thu Jan 05 23:13:02 CST 2017
[INFO] Final Memory: 18M/168M
[INFO] ------------------------------------------------------------------------
```

### CODE REVIEW

File: conf/db/upgrade/V1.9__schema.sql [hunk 1]
```diff
@@ -54,3 +54,273 @@ CREATE TABLE  `zstack`.`AsyncRestVO` (
     `createDate` timestamp,
     PRIMARY KEY  (`uuid`)
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+# BackupStorageEO
+DROP TRIGGER IF EXISTS trigger_clean_AccountResourceRefVO_for_BackupStorageEO;
```
Comment (zxwing) [inline conf/db/upgrade/V1.9__schema.sql]: BackupStroage这些资源是没有AccountResourceRefVO的，这些属于admin资源。只有非admin资源才有AccountResourceRefVO。哪些是非admin资源参考springConfigXml/AccountManager.xml里列表：
```xml
        <property name="resourceTypeForAccountRef">
            <list>
                <value>org.zstack.header.vm.VmInstanceVO</value>
                <value>org.zstack.header.vm.VmNicVO</value>
                <value>org.zstack.header.volume.VolumeVO</value>
                <value>org.zstack.header.image.ImageVO</value>
                <value>org.zstack.header.network.l3.L3NetworkVO</value>
                <value>org.zstack.header.network.l3.IpRangeVO</value>
                <value>org.zstack.network.service.vip.VipVO</value>
                <value>org.zstack.network.securitygroup.SecurityGroupVO</value>
                <value>org.zstack.network.service.portforwarding.PortForwardingRuleVO</value>
                <value>org.zstack.network.service.eip.EipVO</value>
                <value>org.zstack.header.storage.snapshot.VolumeSnapshotVO</value>
                <value>org.zstack.header.identity.UserGroupVO</value>
                <value>org.zstack.header.identity.PolicyVO</value>
                <value>org.zstack.header.identity.UserVO</value>
                <value>org.zstack.header.identity.QuotaVO</value>
                <value>org.zstack.header.configuration.InstanceOfferingVO</value>
                <value>org.zstack.header.configuration.DiskOfferingVO</value>
                <value>org.zstack.network.service.lb.LoadBalancerVO</value>
                <value>org.zstack.network.service.lb.LoadBalancerListenerVO</value>
                <value>org.zstack.header.core.scheduler.SchedulerVO</value>
            </list>
        </property>

```

File: conf/db/upgrade/V1.9__schema.sql [hunk 2]
```diff
@@ -54,3 +54,273 @@ CREATE TABLE  `zstack`.`AsyncRestVO` (
...
+DELIMITER $$
+CREATE TRIGGER trigger_clean_AccountResourceRefVO_for_BackupStorageEO AFTER UPDATE ON `BackupStorageEO`
+FOR EACH ROW
+    BEGIN
+        IF OLD.`deleted` IS NULL AND NEW.`deleted` IS NOT NULL THEN
+            DELETE FROM `AccountResourceRefVO` WHERE `resourceUuid` = OLD.`uuid`;
```
Comment (zxwing) [inline conf/db/upgrade/V1.9__schema.sql]: delete条件要加resourceType = xxx。虽然UUID应该是全局唯一的。但不排除用户在创建不同资源的时候通过API的resourceUuid字段指定了相同的UUID。例如创建VM和创建IMAGE的时候都指定了一样的UUID

Comment (xuexuemiao) [inline conf/db/upgrade/V1.9__schema.sql]: 已添加

File: conf/db/upgrade/V1.9__schema.sql (hunk 1 above)
Comment (xuexuemiao) [inline conf/db/upgrade/V1.9__schema.sql]: 移除了admin的，生成结果统计共计20。与上表数目保持一致。

**Comment by xuexuemiao** [comment]
Successfully applied 15 migrations to schema `zstack` (execution time 00:02.356s).
Flyway 3.2.1 by Boxfuse

Database: jdbc:mysql://localhost:3306/zstack_rest (MySQL 5.5)
Cleaned schema `zstack_rest` (execution time 00:00.004s)
Flyway 3.2.1 by Boxfuse

Database: jdbc:mysql://localhost:3306/zstack_rest (MySQL 5.5)
Validated 1 migration (execution time 00:00.012s)
Creating Metadata table: `zstack_rest`.`schema_version`
Current version of schema `zstack_rest`: << Empty Schema >>
WARNING: outOfOrder mode is active. Migration of schema `zstack_rest` may not be reproducible.
Migrating schema `zstack_rest` to version 0.6 - schema buildin httpserver
Successfully applied 1 migration to schema `zstack_rest` (execution time 00:00.148s).
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 5.491s
[INFO] Finished at: Thu Jan 05 23:49:13 CST 2017
[INFO] Final Memory: 18M/168M
[INFO] ------------------------------------------------------------------------

**Pull request merged**